ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala

```diff
@@ -212,375 +263,243 @@ object LedgerApiErrors extends LedgerApiErrorGroup {
     )
-    @Resolution("Upload the necessary Dars to the participant node.")
-    object MissingPackage
+    @Resolution("Inspect the error message and use a valid offset.")
+    object OffsetOutOfRange
         extends ErrorCode(
-          id = "MISSING_PACKAGE",
-          ErrorCategory.InvalidGivenCurrentSystemStateResourceMissing,
+          id = "OFFSET_OUT_OF_RANGE",
+          ErrorCategory.InvalidGivenCurrentSystemStateOther,
         ) {
-      case class Reject(
-          packageId: PackageId,
-          reference: Reference,
-      )(implicit
+      case class Reject(message: String)(implicit
           loggingContext: ContextualizedErrorLogger
-      ) extends LoggingTransactionErrorImpl(
-            cause = LookupError.MissingPackage.pretty(packageId, reference)
-          )
-
+      ) extends LoggingTransactionErrorImpl(cause = message)
     }
 
     @Explanation(
-      """This error occurs if a package referred to by a Command fails validation. This should not happen as packages are validated when being uploaded."""
+      """Every ledger API command contains a ledger-id which is verified against the running ledger.
+          This error indicates that the provided ledger-id does not match the expected one."""
     )
-    @Resolution("Contact support.")
-    object PackageValidationFailed
-        extends ErrorCode(
-          id = "PACKAGE_VALIDATION_FAILED",
-          ErrorCategory.MaliciousOrFaultyBehaviour,
-        ) {
-      case class Reject(validationErrorCause: String)(implicit
-          loggingContext: ContextualizedErrorLogger
-      ) extends LoggingTransactionErrorImpl(
-            cause = validationErrorCause
-          )
-    }
-  }
-
-  object PreprocessingErrors extends ErrorGroup {
-    @Explanation("""This error occurs if a command fails during interpreter pre-processing.""")
-    @Resolution("Inspect error details and correct your application.")
-    object PreprocessingFailed
-        extends ErrorCode(
-          id = "COMMAND_PREPROCESSING_FAILED",
-          ErrorCategory.InvalidIndependentOfSystemState,
-        ) {
-      case class Reject(
-          err: LfError.Preprocessing.Error
-      )(implicit
-          loggingContext: ContextualizedErrorLogger
-      ) extends LoggingTransactionErrorImpl(
-            cause = err.message
-          )
-    }
-  }
-
-  object InterpreterErrors extends ErrorGroup {
-
-    @Explanation("""This error occurs if the Daml transaction failed during interpretation.""")
-    @Resolution("This error type occurs if there is an application error.")
-    object GenericInterpretationError
+    @Resolution("Ensure that your application is correctly configured to use the correct ledger.")
+    object LedgerIdMismatch
         extends ErrorCode(
-          id = "DAML_INTERPRETATION_ERROR",
+          id = "LEDGER_ID_MISMATCH",
           ErrorCategory.InvalidGivenCurrentSystemStateOther,
```

Comment:
Changed V2 gRPC status code from `RESOURCE_MISSING` to `FAILED_PRECONDITION`

Reply:
Actually, I rushed into changing this. The `FAILED_PRECONDITION` code cannot cover this and `NOT_FOUND` is already a much better proxy.

Description from our docs on `FAILED_PRECONDITION`:
```
The mutable state of the system does not satisfy the preconditions required to execute the request. We consider the whole Daml ledger including ledger config, parties, packages, and command deduplication to be mutable system state. Thus all Daml interpretation errors are reported as FAILED_PRECONDITION errors or one of its specializations.
```